aiida/backends/sqlalchemy/migrations/versions/1de112340b16_django_parity_1.py

```diff
@@ -0,0 +1,173 @@
+        sa.Column('mtime', sa.DateTime(timezone=True)),
+        sa.Column('uuid', UUID(as_uuid=True)),
+    )
+
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.content.is_(None)).values(content=''))
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.mtime.is_(None)).values(mtime=timezone.now()))
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.ctime.is_(None)).values(ctime=timezone.now()))
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.uuid.is_(None)).values(uuid=get_new_uuid()))
+
+    db_dbcomputer = sa.sql.table(
+        'db_dbcomputer',
+        sa.Column('description', sa.Text),
+        sa.Column('hostname', sa.String(255)),
+        sa.Column('metadata', JSONB),
+        sa.Column('scheduler_type', sa.String(255)),
+        sa.Column('transport_type', sa.String(255)),
+        sa.Column('uuid', UUID(as_uuid=True)),
+    )
+
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.description.is_(None)).values(description=''))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.hostname.is_(None)).values(hostname=''))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.metadata.is_(None)).values(metadata={}))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.metadata == JSONB.NULL).values(metadata={}))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.scheduler_type.is_(None)).values(scheduler_type=''))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.transport_type.is_(None)).values(transport_type=''))
+    op.execute(db_dbcomputer.update().where(db_dbcomputer.c.uuid.is_(None)).values(uuid=get_new_uuid()))
+
+    db_dbgroup = sa.sql.table(
+        'db_dbgroup',
+        # sa.Column('user_id', sa.Integer),
+        sa.Column('description', sa.Text),
+        sa.Column('label', sa.String(255)),
+        sa.Column('time', sa.DateTime(timezone=True)),
+        sa.Column('type_string', sa.String(255)),
+        sa.Column('uuid', UUID(as_uuid=True)),
+    )
+
+    op.execute(db_dbgroup.update().where(db_dbgroup.c.description.is_(None)).values(description=''))
+    op.execute(db_dbgroup.update().where(db_dbgroup.c.label.is_(None)).values(label=''))
+    op.execute(db_dbgroup.update().where(db_dbgroup.c.time.is_(None)).values(time=timezone.now()))
+    op.execute(db_dbgroup.update().where(db_dbgroup.c.type_string.is_(None)).values(type_string=''))
+    op.execute(db_dbgroup.update().where(db_dbgroup.c.uuid.is_(None)).values(uuid=get_new_uuid()))
+
+    # db_dbgroup_dbnode = sa.sql.table(
+    #     "db_dbgroup_dbnode",
+    #     # sa.Column('dbgroup_id', sa.Integer),
+    #     # sa.Column('dbnode_id', sa.Integer),
+    # )
+
+    db_dblink = sa.sql.table(
+        'db_dblink',
+        # sa.Column('input_id', sa.Integer),
+        # sa.Column('output_id', sa.Integer),
+        sa.Column('type', sa.String(255)),
+    )
+
+    op.execute(db_dblink.update().where(db_dblink.c.type.is_(None)).values(type=''))
```

Comment:
Already commented on the default setting for the `Comment.ctime`, but this thing seems way more serious. I wonder if we should collect all default settings that would be problematic if there are really rows out there that have null values. We should then have a "integrity-check" that finds these. I fear that if we just silently set these missing values to a default, the database will be broken at other points. Having a `Link` without a type is bad...

Reply:
Well I imagine having them set to `None` would already be bad, and so going from `None -> ''` is not going to break the database any more than it was already.
I think in reality the python code should have already stopped this from occurring, but yeh having an integrity-check could be done

Reply:
Well maybe it would actually be better to have it fail in this case. Assuming that these cases are rare or not existing, we could analyze the problem and provide a proper fix. Tecnhically we would be able to deduce the correct type based on input and output nodes. I think this column is too delicate to be changing silently

Reply:
Ok, so if I just remove this line then.
Is there any other you think should be definitely removed?

I guess we can think to provide a "fix" under `verdi storage` for such issues,
but I imagine we can defer that to another PR?

Reply:
Yes, essentially there are these two comments remaining: (1) remove the setting of defaults that should really be considered as grave bugs (such as this one. Please also check if there are others), and (2) if you keep the commented out lines for operations that should have been there in principle but can't due to problems in original schema, add a generic comment in top docstring.